### PR TITLE
Weather templates from scala

### PR DESCRIPTION
**Pull request by uplne**
Moving templates from JS to Scala + ultimate inline SVGs

Related PRs: 
https://github.com/guardian/platform/pull/688
https://github.com/guardian/fastly-edge-cache/pull/395

@robertberry @sammorrisdesign @gklopper

### CODE REVIEW

File: static/src/stylesheets/module/_weather.scss [hunk 1]
```diff
@@ -104,7 +104,7 @@ $weather-large-size: 60px;
     }
 }
 
-.i.weather__icon {
+.weather__icon svg {
```
Comment (sndrs) [inline static/src/stylesheets/module/_weather.scss]: does this need to go on the svg itself? can it just go on `.weather__icon`?

File: static/src/stylesheets/module/_weather.scss [hunk 2]
```diff
@@ -306,12 +306,12 @@ i.weather__close-icon {
         }
     }
 
-    &.forecast__item--1 {
+    &.forecast__item--0 {
```
Comment (sammorrisdesign) [inline static/src/stylesheets/module/_weather.scss]: Stupid 0 indexing...

File: weather/app/views/cityWeather.scala.html [hunk 3]
```diff
@@ -1,29 +1,35 @@
...
             <span class="u-h">The temperature</span>
             <span class="weather__time">Now</span>
             <span class="u-h">is</span>
-            <span class="weather__temp js-weather-temp">{{tempNow}}</span>
+            @defining(weather.temperatureForEdition(Edition(request))) { temp =>
+                <span class="weather__temp js-weather-temp">@math.round(temp.Value)°@temp.Unit</span>
+            }
         </p>
-        <i class="i i-weather-{{icon}} weather__icon js-weather-icon" title="{{description}}"><span class="u-h">{{description}}</span></i>
+        @fragments.inlineSvg(s"weather-${weather.WeatherIcon}", "weather", Seq("weather__icon","js-weather-icon"))<span class="u-h">@weather.WeatherText</span>
```
Comment (sndrs) [inline weather/app/views/cityWeather.scala.html]: do you need the `.weather__icon` class? you get `inline-weather` by default

File: static/src/stylesheets/module/_weather.scss (hunk 1 above)
Comment (uplne) [inline static/src/stylesheets/module/_weather.scss]: It's mainly because svg is setting w/h of the icons to 32px as default and we want to change the size. I can use '.weather__icon svg' just to change the size of the SVG and leave everything else to just '.weather__icon' but it will just add another rule so I decided to do it all directly on svg.

Do we actually need that 'span' wrapping the svg? Can we add classes right to the SVG. I suppose it must be at more places where we actually want to change the size of the SVGs.

File: weather/app/views/cityWeather.scala.html (hunk 3 above)
Comment (uplne) [inline weather/app/views/cityWeather.scala.html]: I would rather use '.weather__icon', as it fits the structure of the component, and remove the 'inline-weather'.

**Comment by gklopper** [comment]
We moving on this?

File: static/src/stylesheets/module/_weather.scss (hunk 1 above)
Comment (sndrs) [inline static/src/stylesheets/module/_weather.scss]: fair enough.

i didn't want that span, it was a performance issue in scala (having to parse the svg vs stuff the output wrapped in a span). I'd like to find a way to remove it.

File: weather/app/views/cityWeather.scala.html (hunk 3 above)
Comment (sndrs) [inline weather/app/views/cityWeather.scala.html]: when everything is over, `.inline-` will become `.i-` to match the current style. don't knwo if that makes a diff.

File: static/src/javascripts/projects/facia/modules/onwards/weather.js [hunk 4]
```diff
@@ -232,61 +227,28 @@ define([
...
-                this.toggleControls();
+                this.attachToDOM(weatherData.html, city);
+                searchTool.bindElements($('.js-search-tool'));
             };
         },
 
+        attachToDOM: function (tmpl, city) {
+            $holder = $('.js-container--first .js-container__header');
+            $holder.empty();
+            $holder.html(tmpl.replace(new RegExp('{{city}}', 'g'), city));
```
Comment (phamann) [inline static/src/javascripts/projects/facia/modules/onwards/weather.js]: Should now wrap this is a `fastdom.write()` method call.

File: static/src/javascripts/projects/facia/modules/onwards/weather.js [hunk 5]
```diff
@@ -232,61 +227,28 @@ define([
...
-                }));
-
-                docFragment.appendChild($forecast[0]);
-            }
+                tmpl            = forecastData.html;
 
-            $forecastHolder.each(function (item) {
-                $(item).append(docFragment.cloneNode(true));
-            });
+            $forecastHolder.empty().html(tmpl);
```
Comment (phamann) [inline static/src/javascripts/projects/facia/modules/onwards/weather.js]: AS above, wrap in `fastdom.write`

File: weather/app/views/cityForecast.scala.html [hunk 6]
```diff
@@ -0,0 +1,17 @@
+@(forecasts: Seq[models.ForecastResponse])(implicit request: RequestHeader)
+
+@import common.Edition
+
+@for((forecast, index) <- forecasts.zipWithIndex) {
+    <li class="forecast__item forecast__item--@index">
+        <p class="weather__desc">
+            <span class="u-h">The temperature at</span>
+            <span class="weather__time">@forecast.hourString</span>
```
Comment (phamann) [inline weather/app/views/cityForecast.scala.html]: Did you consider using a `<dl>` for this instead? Also would be nice to see some schema here too and the time wrapped in a `<time>` element.

**Comment by phamann** [comment]
:+1: Other than my comments. Also looks like you've got some merge conflicts.

File: static/src/javascripts/projects/facia/modules/onwards/weather.js (hunk 4 above)
Comment (uplne) [inline static/src/javascripts/projects/facia/modules/onwards/weather.js]: I would like to do all fastdom updates after this will be deployed as separate PR. There are more places that can use it around weather js.

**Comment by michaelwmcnamara** [comment]
To be tested on mobile devices also - specifically:
1) what happens if you tap elsewhere on page while entering text
2) what happens when you close keyboard while entering text
